### PR TITLE
setup.py to create a local installation, test and clean

### PR DESCRIPTION
**Pull request by omegaliii**
Support install, test, clean using setup.py locally.

Instruction:
```
python3 setup.py install
python3 setup.py test --ip=${NFS4_SERVER_IP}
python3 setup.py clean
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    
+    def run(self):
+        test_loader = unittest.TestLoader()
+        test_suite = test_loader.discover('test', pattern='sfs_test.py')
+        
+        import sfs
+        try:
+            sfs.mount(self.ip)
+        except Exception as e:
+            print("NFS SSERVER IP: ", self.ip)
```
Comment (jssmith) [inline setup.py]: Extra S in server. Also, please enhance the error message to provide guidance on how to fix the problem.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        subprocess.call(['make', 'clean'], cwd = NFSV4_PATH)
+        subprocess.call(['rm', 'libnfs4.so'], cwd = SFS_PATH)
+        os.system('rm -vrf ./build ./dist ./*.pyc ./*.tgz ./*.egg-info')
```
Comment (jssmith) [inline setup.py]: We have some checked-in files in `.dist`. Does `setup.py` put anything there? If not then we don't need to clean up there. If so we should move away what is already in dist (an old example that probably shouldn't be checked in) to another path.

**Review by jssmith — COMMENTED** [review]
This looks really good overall. Please see some of the specific comments. Also, please update `README.md` with instructions for building.

File: setup.py (hunk 2 above)
Comment (omegaliii) [reply setup.py]: setup.py will only generate build and .egg-info. So running clean should only remove build and .egg-info right now.

**Review by jssmith — APPROVED** [review]
This looks good to me.

**Pull request merged**